### PR TITLE
Update continuous integration workflow configurations.

### PR DESCRIPTION
**Pull request by xydesa**

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -14,34 +14,38 @@ jobs:
...
-        python-version: ['3.7', '3.8', '3.9', '3.10']
+        os: [macos-latest, ubuntu-20.04, ubuntu-22.04, windows-latest]
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
```
Comment (denewiler) [inline .github/workflows/tests.yaml:24]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -65,14 +69,8 @@ jobs:
...
-        if: matrix.os == 'ubuntu-18.04' || matrix.os == 'ubuntu-20.04'
+        if: matrix.os == 'ubuntu-20.04' || matrix.os == 'ubuntu-22.04'
         run: |
-          sudo apt install curl
-          curl -sL https://deb.nodesource.com/setup_14.x -o nodesource_setup.sh
-          sudo bash nodesource_setup.sh
-          sudo apt install nodejs
-          sudo npm install -g n
-          sudo n stable
           sudo npm install -g eslint
```
Comment (denewiler) [inline .github/workflows/tests.yaml:74]:
```suggestion
          npm install -g eslint
```

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -87,7 +85,6 @@ jobs:
           sudo npm install -g postcss-html # needed for stylelint to process html
           sudo npm install -g stylelint
           sudo npm install -g stylelint-config-standard
```
Comment (denewiler) [inline .github/workflows/tests.yaml:87]:
```suggestion
          npm install -g postcss-html # needed for stylelint to process html
          npm install -g stylelint
          npm install -g stylelint-config-standard
```

File: tox.ini [hunk 4]
```diff
@@ -28,6 +28,7 @@ python =
     3.8: py38
     3.9: py39
     3.10: py310
+    3.11: py311
```
Comment (denewiler) [inline tox.ini:31]: Line 2, `envlist` needs `py311 appended.

**Review by denewiler — CHANGES_REQUESTED** [review]

**Comment by xydesa** [comment]
@denewiler it seems like the issue with the failing test may be related to this ticket in stylelint: https://github.com/stylelint/stylelint/issues/6525
fixed in this PR: https://github.com/stylelint/stylelint/pull/6547

does that make sense to you?

If so, we may need to wait for a new release.

**Comment by xydesa** [comment]
> @denewiler it seems like the issue with the failing test: https://github.com/sscpac/statick-web/actions/runs/4048849626/jobs/6964571593 may be related to this ticket in stylelint: [stylelint/stylelint#6525](https://github.com/stylelint/stylelint/issues/6525) fixed in this PR: [stylelint/stylelint#6547](https://github.com/stylelint/stylelint/pull/6547)
> 
> does that make sense to you?
> 
> If so, we may need to wait for a new release of stylelint.

nvermind, it looks like it might be more related to this: https://github.com/stylelint/stylelint/issues/1973

also, switching to the same local installation as the estlint configs seems to have fixed the issue.

**Pull request closed without merging**